porch/pkg/cache/draft.go

```diff
@@ -31,6 +31,11 @@ func (cd *cachedDraft) Close(ctx context.Context) (repository.PackageRevision, e
 	if closed, err := cd.PackageDraft.Close(ctx); err != nil {
 		return nil, err
 	} else {
-		return cd.cache.update(closed)
+		updated, err := cd.cache.update(closed)
+		if err != nil {
+			return nil, err
+		}
+		_, _, err = cd.cache.refreshAllCachedPackages(ctx)
```

Comment:
The reason was that because the package revision key now changes when it is published (since it gets assigned a revision), the old cache was invalid as it had the wrong key stored in it. But I think you're right that refreshing the whole cache isn't necessary. I instead modified the `update` code in `porch/pkg/cache/repository.go` to delete the old key from the cache before adding the new one, which I think equally fixes the issue.